opsramp/base.py

```diff
@@ -121,6 +122,62 @@ def chroot(self, suffix=''):
             self.tracker.reset()
         return self.compute_url()
 
+    @staticmethod
+    def collate_pages(get_request, data):
+        """Given a GET request whose results span across multiple pages, crawl
+        each page and collate the results.
+
+        :param first_page_data: "results" dict for first pagefull of dat
+        :type first_page_data: dict
+        :param request: Request used to get first page
+        :type request: requests.PreparedRequest
+        """
+        # First, sanity check that all is good. Only process GET requests:
+        if get_request.method.upper().strip() != "GET":
+            return data
+
+        collated_data = data["results"]
+
+        # Only attempt to pull subsequent pages if we can verify that there are
+        # subsequent pages "to be pulled"...
+        if "results" in data.keys():
+            while "nextPage" in data.keys() and data["nextPage"]:
+                # Get the next page full of data.
+                next_page = requests.get(get_request.url,
+                                         params={'pageNo': data['nextPageNo']},
+                                         headers=get_request.headers)
+
+                if not next_page.ok:
+                    # At least return what we can.
+                    return collated_data
+
+                data = next_page.json()
```

Comment:
.json() method had a try wrapped round it in the original code?

Reply:
That's a "for testing" oversight. Will undo.